server/ast/index_elem.go

```diff
@@ -0,0 +1,64 @@
+// Copyright 2023 Dolthub, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package ast
+
+import (
+	"fmt"
+
+	vitess "github.com/dolthub/vitess/go/vt/sqlparser"
+
+	"github.com/dolthub/doltgresql/postgres/parser/sem/tree"
+)
+
+// nodeIndexElemList converts a tree.IndexElemList to a slice of vitess.IndexColumn.
+func nodeIndexElemList(node tree.IndexElemList) ([]*vitess.IndexColumn, error) {
```

Comment:
This should take the functionality out of here:
https://github.com/dolthub/doltgresql/blob/6da1bf1992806e324db7b7869e83d0c0ffa45a56/server/ast/index_table_def.go#L41-L83
It's the exact same thing, but we didn't split it up into different files since we didn't need to at the time.

Reply:
Perfect! Thanks for pointing out that existing code. Fixed in 2a516b566c64cd1c9a8764b210ef2d8003fa344c